### PR TITLE
camera: Improve error handle

### PR DESCRIPTION
**Pull request by patrickelectric**
Return error over unwrap to help with get-all

### CODE REVIEW

File: onvif/examples/camera.rs [hunk 1]
```diff
@@ -207,11 +208,9 @@ async fn get_system_date_and_time(clients: &Clients) {
     println!("{:#?}", date);
 }
 
-async fn get_stream_uris(clients: &Clients) {
+async fn get_stream_uris(clients: &Clients) -> Result<(), transport::Error> {
     let media_client = clients.media.as_ref().unwrap();
```
Comment (chrisabruce) [inline onvif/examples/camera.rs]: If we are returning a result, as good practice, we should return all the errors, not just the transport one and let the other unwrap.

File: onvif/examples/camera.rs [hunk 2]
```diff
@@ -268,25 +265,25 @@ async fn get_hostname(clients: &Clients) {
...
     )
-    .await
-    .unwrap();
+    .await?;
+    Ok(())
 }
 
-async fn enable_analytics(clients: &Clients) {
+async fn enable_analytics(clients: &Clients) -> Result<(), transport::Error> {
     let media_client = clients.media.as_ref().unwrap();
```
Comment (chrisabruce) [inline onvif/examples/camera.rs]: Another unwrap not handled by return result.

File: onvif/examples/camera.rs [hunk 3]
```diff
@@ -344,27 +338,27 @@ async fn enable_analytics(clients: &Clients) {
...
             profiles.profiles.len()
         );
     }
+    Ok(())
 }
 
-async fn get_analytics(clients: &Clients) {
+async fn get_analytics(clients: &Clients) -> Result<(), transport::Error> {
     let config = schema::media::get_video_analytics_configurations(
         clients.media.as_ref().unwrap(),
```
Comment (chrisabruce) [inline onvif/examples/camera.rs]: Again, probably should handle this unrwap since we are returning a result.

**Review by chrisabruce — DISMISSED** [review]

**Comment by jplatte** [comment]
Ping @patrickelectric, are you planning to continue work on this?

**Comment by patrickelectric** [comment]
@jplatte, Sorry, I'm a bit out of time. If someone wants to continue or rework it, be free.

**Comment by patrickelectric** [comment]
@jplatte @chrisabruce, patch updated

**Comment by jplatte** [comment]
@patrickelectric can you rebase so CI will (hopefully) succeed? The other PR was trivial so I merged it without a rebase, but this one isn't so I think we should make sure there's no new issues. Also needs a review from sbd. with onvif knowledge of course...

File: onvif/examples/camera.rs [hunk 4]
```diff
@@ -257,13 +262,19 @@ async fn get_stream_uris(clients: &Clients) {
...
-        .unwrap();
+async fn get_snapshot_uris(clients: &Clients) -> Result<(), transport::Error> {
+    let media_client = match clients.media.as_ref() {
+        Some(client) => client,
+        None => {
+            return Err(transport::Error::Other(
+                "Client media is not available".into(),
+            ))
+        }
+    };
```
Comment (DmitrySamoylov) [inline onvif/examples/camera.rs]: This `match` can be replaced with `.ok_or(..)?` here and below.

Comment (jplatte) [reply onvif/examples/camera.rs]: Probably `ok_or_else` since the `.into` allocates and clippy will complain otherwise.

Comment (patrickelectric) [reply onvif/examples/camera.rs]: Good call!

Comment (patrickelectric) [reply onvif/examples/camera.rs]: In the end I'm using `.ok_or` since it makes a bit more clear the code.

Comment (jplatte) [reply onvif/examples/camera.rs]: Like I was saying, now clippy complains and I would tend to agree that its suggestion makes sense. Can you please apply them? (can be automated via `cargo clippy --fix`)

**Review by jplatte — APPROVED** [review]

**Pull request merged**